src/hott/types/2_adj/two_adj.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Authors: Daniel Carranza, Jonathon Chang, Ryan Sandford
+Under the supervision of Chris Kapulkin
+
+Theorems about two half-adjoint equivalences, 
+  including a proof that two left half-adjoint
+  and two right half-adjoint equivalences are propositions
+  and that the two full-adjoint equivalence is equivalent to a non-propositional type 
+
+Last updated: 2020-06-12
+-/
+
+import hott.init hott.types.sigma hott.types.prod hott.types.pi hott.types.fiber hott.types.equiv .adj .prelim .hty_rec
+open hott hott.eq
+
+hott_theory
+universes u v
+
+-- This is used in the definition of the compatibilites for
+--   a two half-adjoint and two left half-adjoint equivalence
+@[hott] def nat_coh {A : Type u} {B : Type _} (g : B → A) (f : A → B) (H : g ∘ f ~ id) (x : A) 
+  : H (g (f x)) = ap g (ap f (H x)) :=
+cancel_right (H x) (ap_con_eq_con H (H x))⁻¹ ⬝ ap_compose g f _
+
+namespace equiv
+    variables {A B: Type u}
+
+    -- Right coherence for two half-adjoint equivalence
+    @[hott] def r2coh (f : A → B) (h : adj f) (x : A) :=
+    nat_coh h.1 f h.2.1 x ⬝ ap02 h.1 (h.2.2.2.1 x) = h.2.2.2.2 (f x)
+
+    -- Left coherence for two half-adjoint equivalence
+    @[hott] def l2coh (f : A → B) (h : adj f) (y : B) :=
+    h.2.2.2.1 (h.1 y) ⬝ (nat_coh f h.1 h.2.2.1 y) = ap02 f (h.2.2.2.2 y)
+
+    -- Definition of a two (right) half-adjoint equivalence
+    @[hott] def is_two_hae (f : A → B) :=
```

Comment:
It's probably useful to define projections for this (and `is_two_hae_l` / `two_adj` / `is_hadj_l`), too.